### PR TITLE
Add libretro-common as submodule

### PR DESCRIPTION
**Pull request by LEGOAnimal22**
Included the libretro solution as a submodule within the project, as suggested in #196.
I confirmed that this change both builds and runs.

### CODE REVIEW

File: frontend/libretro.c [hunk 1]
```diff
@@ -31,7 +31,7 @@
 #include "plugin_lib.h"
 #include "arm_features.h"
 #include "revision.h"
-#include "libretro.h"
+#include "libretro-common/include/libretro.h"
```
Comment (RobLoach) [inline frontend/libretro.c]: A cleaner solution would be to add `libretro-common/include` to the include directories in Makefile.common. Then just use `#include "libretro.h"`, and the compiler will file it correctly.

File: .gitmodules [hunk 2]
```diff
@@ -0,0 +1,3 @@
+[submodule "frontend/libretro-common"]
+	path = frontend/libretro-common
```
Comment (RobLoach) [inline .gitmodules]: Any reason why this is in `frontend`? When I've seen libretro-common in cores, they've just been in the root with `/libretro-common`.

**Comment by orbea** [comment]
> Are you suggesting bringing in the history of RetroArch for the subtree of libretro-common?

No. I'm not sure where you got that idea, I'm talking about the git feature called subtree, its what RetroArch uses for gslang and maybe some other directories.

https://www.atlassian.com/blog/git/alternatives-to-git-submodule-git-subtree

The problems with git submodules is that it adds another build step for any users trying to build cores and makes the clone slower when it clones the entire history of libretro-common. This is significant in some parts of the world with limited internet access. It also makes it harder to package tarballs from git commits which no longer have libretro-common or any `.git` directory required to clone them.

There are a lot of other reasons and this has been discussed on irc in the past.

> Twinaphex updates the libretro-common repository periodically. I see more complexity out of using the git subtree than the submodule. What issue do you have with submodules?

That said I'm fine with the current method and this PR doesn't fix any real issues.

**Comment by bparker06** [comment]
Submodules cannot be modified... with subtrees you can at least make your own local commits.

While I generally like subtrees better I don't think either is a viable solution for libretro-common because most cores only use a subset of libretro-common... and any upstream devs would quite likely get very angry at bringing in the entirety of libretro-common for no reason... even if it's a submodule where it's not technically inside the repo.

Also if the libretro-common upstream repo ever disappeared or moved, having that as a submodule would instantly break the core for everyone. Subtree and/or manual file management solves that particular futureproofing issue IMO.

**Comment by LEGOAnimal22** [comment]
> Submodules cannot be modified... with subtrees you can at least make your own local commits.

What's to modify? The only file from `libretro-common` that this repo depends on is the `libretro.h` header file, and the version of the header that is kept here seems to reflect the `libretro-common` one. If anything, making `libretro-common` a submodule saves some time by using the latest version of the header directly instead of having to manually copy over individual updates.

**Comment by orbea** [comment]
> What's to modify?

Modifying `libretro-common` to test core specific features like chd support is a pretty common occurrence and making is a submodule will make that more tedious. Note that it may not be desirable to merge these changes everywhere until the feature is debugged and tested.

Also @bparker06 is very right that the repo can potentially move in the future and that will certainly break many repos...

**Comment by RobLoach** [comment]
I think this is only being used for `libretro.h` at the moment. No real need currently, unless pcsx_rearmed ends up adopting a lot more of libretro-common.

**Comment by orbea** [comment]
Good point, but I was talking more generally than this one repo since I am under the impression that the suggestion is also more general? Though I suppose pcsx_rearmed might adopt more of libretro-common in the future, kind of hard to predict that. :)

**Comment by bparker06** [comment]
If it's only being used for libretro.h for now, I would just commit that file manually like the other cores do. Frontends have to be backwards compatible with the libretro API version anyway, so having an old version won't break anything.

**Comment by m4xw** [comment]
Late to the party, but if at all, only [git subrepo](https://github.com/ingydotnet/git-subrepo) is acceptable, IMO

**Comment by retro-wertz** [comment]
i dont think we need anything other than libretro.h at the moment....

**Comment by orbea** [comment]
I'm in favor of either removing the unneeded files or just ignoring them until they are actually used since they don't really cause any problems.

**Comment by retro-wertz** [comment]
> I'm in favor of either removing the unneeded files or just ignoring them until they are actually used since they don't really cause any problems.

i actually meant the purpose of the PR in which case we dont need anything on the libretro-common of things. 

About clean-up, i thought this was suppose to be upstream-friendly...

**Comment by orbea** [comment]
> About clean-up, i thought this was suppose to be upstream-friendly...

I was talking about libretro-common only, removing unneeded files from libretro-common is not really a big deal.

**Comment by retro-wertz** [comment]
there is no libretro-common in this core.. only libretro.h is used regardless...

**Comment by orbea** [comment]
I stand corrected, then this PR is entirely pointless?

**Comment by RobLoach** [comment]
Its aim is sound, but not really needed right now, as it's just used for `libretro.h`. Copy-Pasta is easiest.

**Pull request closed without merging**